pennylane/grouping/utils.py

```diff
@@ -333,6 +335,245 @@ def binary_to_pauli(binary_vector, wire_map=None):  # pylint: disable=too-many-b
+    if wire_map is None:
+        wire_map = {pauli_word.wires.labels[i]: i for i in range(len(pauli_word.wires))}
+
+    n_qubits = len(wire_map)
+
+    # If there is only a single qubit, we can return the matrix directly
+    if n_qubits == 1:
+        return pauli_word.matrix
+
+    # There may be more than one qubit in the Pauli but still only
+    # one of them with anything acting on it, so take that into account
+    pauli_names = [pauli_word.name] if isinstance(pauli_word.name, str) else pauli_word.name
+
+    # Special case: the identity Pauli
+    if pauli_names == ["Identity"]:
+        return np.eye(2 ** n_qubits)
+
+    # If there is more than one qubit, we must go through the wire map wire
+    # by wire and pick out the relevant matrices
+    pauli_mats = [np.eye(2) for x in range(n_qubits)]
+
+    for wire_label, wire_idx in wire_map.items():
+        if wire_label in pauli_word.wires.labels:
+            op_idx = pauli_word.wires.labels.index(wire_label)
+            pauli_mats[wire_idx] = getattr(qml, pauli_names[op_idx]).matrix
+
+    return reduce(np.kron, pauli_mats)
+
+
+def is_commuting(pauli_word_1, pauli_word_2, wire_map=None):
+    r"""Checks if two Pauli words commute.
+
+    To determine if two Pauli words commute, we can check the value of the
+    symplectic inner product of their binary vector representations.
+    For two binary vectors representing Pauli words, :math:`p_1 = [x_1, z_1]`
+    and :math:`p_2 = [x_2, z_2],` the symplectic inner product is defined as
+    :math:`\langle p_1, p_2 \rangle_{symp} = z_1 x_2^T + z_2 x_1^T`. If the symplectic
+    product is 0 they commute, while if it is 1, they don't commute.
+
+    Args:
+        pauli_word_1 (Observable): first Pauli word in commutator
+        pauli_word_2 (Observable): second Pauli word in commutator
+        wire_map (dict[Union[str, int], int]): dictionary containing all wire labels used in
+            the Pauli word as keys, and unique integer labels as their values
+
+    Returns:
+        bool: returns True if the input Pauli commute, False otherwise
+
+    Raises:
+        TypeError: if either of the Pauli words is not valid.
+
+    **Example**
+
+    >>> wire_map = {'a' : 0, 'b' : 1, 'c' : 2}
+    >>> pauli_word_1 = qml.PauliX('a') @ qml.PauliY('b')
+    >>> pauli_word_2 = qml.PauliZ('a') @ qml.PauliZ('c')
+    >>> is_commuting(pauli_word_1, pauli_word_2, wire_map=wire_map)
+    False
+    """
+
+    if not (is_pauli_word(pauli_word_1) and is_pauli_word(pauli_word_2)):
+        raise TypeError(
+            "Expected Pauli word observables, instead got {} and {}".format(
+                pauli_word_1, pauli_word_2
+            )
+        )
+
+    # If no wire map is specified, generate one from the union of wires
+    # in both Paulis.
+    if wire_map is None:
+        wire_labels = Wires.all_wires([pauli_word_1.wires, pauli_word_2.wires]).labels
+        wire_map = {label: i for i, label in enumerate(wire_labels)}
```

Comment:
These lines seem to be repeated in multiple functions, maybe worth creating a small utility function such that code will need to change at a single place (also the comment can become the docstring of the function).